### PR TITLE
Add httpConnectTimeout and httpReadTimeout to google config

### LINKED ISSUE #3973: Google batch timeout - add google.batch.storage.httpRequestTimeout configuration option
## New feature
Currently, when running pipelines in Google batch  larger (>1gb) intermediary files not get transferred over to the outdir.

The current default timeout is 20 Seconds and the fix seems to be to increase the timeout: https://github.com/googleapis/google-cloud-java/issues/2243#issuecomment-320751395 

I do not see a configuration in Nextflow to extend the current timeout: google.batch.storage.httpRequestTimeout. 

Spoke with Jordi and using Fusion will not help in this case either since its Nextflow doing the copying to the publishDir. 

```
java.io.IOException: All 0 retries failed. Waited a total of 0 ms between attempts
  at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.asIoException(CloudStorageFileSystemProvider.java:1052)
  at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.copy(CloudStorageFileSystemProvider.java:691)
  at nextflow.file.FileHelper.copyPath(FileHelper.groovy:959)
  at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:466)
  at nextflow.processor.PublishDir.processFile(PublishDir.groovy:377)
  at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:359)
  at jdk.internal.reflect.GeneratedMethodAccessor524.invoke(Unknown Source)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
  at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
  at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1258)
  at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
  at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1036)
  at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1019)
  at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:97)
  at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:332)
  at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
  at groovy.lang.Closure.run(Closure.java:493)
  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.google.cloud.storage.StorageException: All 0 retries failed. Waited a total of 0 ms between attempts
  at com.google.cloud.storage.contrib.nio.CloudStorageRetryHandler.handleRetryForStorageException(CloudStorageRetryHandler.java:135)
  at com.google.cloud.storage.contrib.nio.CloudStorageRetryHandler.handleStorageException(CloudStorageRetryHandler.java:115)
  at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.copy(CloudStorageFileSystemProvider.java:688)
  ... 22 common frames omitted
Caused by: com.google.cloud.storage.StorageException: Read timed out
  at com.google.cloud.storage.StorageException.translate(StorageException.java:118)
  at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:287)
  at com.google.cloud.storage.spi.v1.HttpStorageRpc.rewrite(HttpStorageRpc.java:1123)
  at com.google.cloud.storage.spi.v1.HttpStorageRpc.openRewrite(HttpStorageRpc.java:1053)
  at com.google.cloud.storage.StorageImpl.lambda$copy$22(StorageImpl.java:515)
  at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
  at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
  at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
  at com.google.cloud.storage.Retrying.run(Retrying.java:54)
  at com.google.cloud.storage.StorageImpl.run(StorageImpl.java:1406)
  at com.google.cloud.storage.StorageImpl.copy(StorageImpl.java:513)
  at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.copy(CloudStorageFileSystemProvider.java:682)
  ... 22 common frames omitted
Caused by: java.net.SocketTimeoutException: Read timed out
  at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:283)
  at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:309)
  at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
  at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
  at java.base/java.net.Socket$SocketInputStream.read(Socket.java:966)
  at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:484)
  at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:478)
  at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
  at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1465)
  at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1069)
  at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)
  at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
  at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:343)
  at java.base/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:824)
  at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:759)
  at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1688)
  at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
  at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
  at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
  at com.google.api.client.http.javanet.NetHttpResponse.<init>(NetHttpResponse.java:36)
  at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:152)
  at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:84)
  at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1012)
  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:525)
  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:466)
  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:576)
  at com.google.cloud.storage.spi.v1.HttpStorageRpc.rewrite(HttpStorageRpc.java:1113)
  ... 31 common frames omitted
```

### PR DESCRIPTION
**Pull request by bentsherman**
Close #3973

### CODE REVIEW

File: plugins/nf-amazon/src/main/nextflow/cloud/aws/config/AwsS3Config.groovy [hunk 1]
```diff
@@ -50,8 +50,8 @@ class AwsS3Config {
         this.debug = opts.debug as Boolean
         this.endpoint = opts.endpoint ?: SysEnv.get('AWS_S3_ENDPOINT')
         this.storageClass = parseStorageClass((opts.storageClass ?: opts.uploadStorageClass) as String)     // 'uploadStorageClass' is kept for legacy purposes
-        this.storageEncryption = parseStorageEncryption(opts.storageEncryption as String)
-        this.storageKmsKeyId = opts.storageKmsKeyId
+        this.storageEncryption = parseStorageEncryption(opts.storageEncryption as String) ?: SysEnv.get('NXF_AWS_SSE_MODE')
+        this.storageKmsKeyId = opts.storageKmsKeyId ?: SysEnv.get('NXF_AWS_SSE_KMS_KEY_ID')
```
Comment (pditommaso) [inline plugins/nf-amazon/src/main/nextflow/cloud/aws/config/AwsS3Config.groovy:54]: This should not be here

File: plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy [hunk 2]
```diff
@@ -121,4 +121,18 @@ class AwsS3ConfigTest extends Specification {
...
+
+        when:
+        def client = new AwsS3Config([:])
+        then:
+        client.storageKmsKeyId == 'xyz1'
+        client.storageEncryption == 'aws:kms'
+
+        cleanup:
+        SysEnv.pop()
+    }
```
Comment (pditommaso) [inline plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy:137]: also this looks unrelated

File: docs/config.md [hunk 3]
```diff
@@ -715,6 +715,16 @@ The following settings are available:
...
 
+`google.httpConnectTimeout`
+: :::{versionadded} 23.06.0-edge
+  :::
+: Defines the HTTP connection timeout for Cloud Storage API requests (default: `'20s'`).
+
+`google.httpReadTimeout`
+: :::{versionadded} 23.06.0-edge
+  :::
+: Defines the HTTP read timeout for Cloud Storage API requests (default: `'20s'`).
```
Comment (pditommaso) [inline docs/config.md:726]: Are these the stock defaults? it may be better to increase them

**Review by pditommaso — CHANGES_REQUESTED** [review]

File: plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy (hunk 2 above)
Comment (bentsherman) [reply plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy:137]: Did you force push to master? Maybe that messed up the PR branch

File: docs/config.md (hunk 3 above)
Comment (bentsherman) [reply docs/config.md:726]: They are. and yes they should probably be higher. Not sure what the AWS defaults are but I bet they are higher

File: plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy (hunk 2 above)
Comment (pditommaso) [reply plugins/nf-amazon/src/test/nextflow/cloud/aws/config/AwsS3ConfigTest.groovy]: yes, that was the reason

**Comment by pditommaso** [comment]
Many tests fails with this error 

```
Unable to load cache item
java.lang.IllegalStateException: Unable to load cache item
	at net.sf.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:79)
	at net.sf.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:119)
	at net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:294)
	at net.sf.cglib.proxy.Enhancer.createHelper(Enhancer.java:480)
	at net.sf.cglib.proxy.Enhancer.createClass(Enhancer.java:337)
	at org.spockframework.mock.runtime.CglibMockFactory.createMock(CglibMockFactory.java:34)
	at org.spockframework.mock.runtime.ProxyBasedMockFactory.create(ProxyBasedMockFactory.java:45)
	at org.spockframework.mock.runtime.JavaMockFactory.createInternal(JavaMockFactory.java:58)
	at org.spockframework.mock.runtime.JavaMockFactory.create(JavaMockFactory.java:38)
	at org.spockframework.mock.runtime.CompositeMockFactory.create(CompositeMockFactory.java:44)
	at org.spockframework.lang.SpecInternals.createMock(SpecInternals.java:50)
	at org.spockframework.lang.SpecInternals.createMockImpl(SpecInternals.java:305)
	at org.spockframework.lang.SpecInternals.createMockImpl(SpecInternals.java:295)
	at org.spockframework.lang.SpecInternals.MockImpl(SpecInternals.java:108)
	at nextflow.cloud.google.lifesciences.GoogleLifeSciencesHelperTest.$spock_initializeFields(GoogleLifeSciencesHelperTest.groovy:36)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/java.lang.Class.forName(Class.java:495)
	at java.base/java.lang.Class.forName(Class.java:474)
	at net.sf.cglib.core.ReflectUtils.defineClass(ReflectUtils.java:467)
	at net.sf.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:339)
	at net.sf.cglib.proxy.Enhancer.generate(Enhancer.java:492)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:96)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:94)
	at net.sf.cglib.core.internal.LoadingCache$2.call(LoadingCache.java:54)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at net.sf.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:61)
	... 15 more
Caused by: java.lang.IllegalStateException: You are currently running with version 2.2.0 of google-api-client. You need at least version 1.31.1 of google-api-client to run version 1.31.5 of the Cloud Life Sciences API library.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:534)
	at com.google.api.client.util.Preconditions.checkState(Preconditions.java:113)
	at com.google.api.services.lifesciences.v2beta.CloudLifeSciences.<clinit>(CloudLifeSciences.java:44)
	... 25 more
```

File: docs/config.md (hunk 3 above)
Comment (bentsherman) [reply docs/config.md]: Okay I increased the defaults to 60 seconds

**Comment by pditommaso** [comment]
But some specific API are used from that library, right? If so it's better to include an explicit deps

**Comment by bentsherman** [comment]
Fair enough, I tested with the explicit dependency and it also works, see the second-to-last commit. I tried to trick github by force-removing the last commit, but it's running the tests again anyway
